internal/service/eks/addon_version_data_source.go

```diff
@@ -0,0 +1,59 @@
+package eks
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+)
+
+func DataSourceAddonVersion() *schema.Resource {
+	return &schema.Resource{
+		ReadWithoutTimeout: dataSourceAddonVersionRead,
+		Schema: map[string]*schema.Schema{
+			"addon_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"kubernetes_version": {
```

Comment:
We often include a `filter` attribute for data sources.

Reference:

- Data Source: https://github.com/hashicorp/terraform-provider-aws/blob/a65002b873e17df5d0f19a6d16ff53d0c91a5aab/internal/service/ec2/ebs_snapshot_data_source.go#L38
- Test File: https://github.com/hashicorp/terraform-provider-aws/blob/a65002b873e17df5d0f19a6d16ff53d0c91a5aab/internal/service/ec2/ebs_snapshot_data_source_test.go#L125

Reply:
Yes thats right. However do you think it really makes sense for this datasource which computes a single version string? There are not many ways to filter the returnable string. The arguments to the datasource, `addon_name`, `kubernetes_version` and `most_recent` are all and exactly what we need to get at a specific version.

Reply:
> The arguments to the datasource, addon_name, kubernetes_version and most_recent are all and exactly what we need to get at a specific version.

You might want to add `platform_version` too, see https://github.com/aws/aws-cli/issues/6832